### PR TITLE
Replace file opening with `with open() as`

### PR DESCRIPTION
**Pull request by KochTobi**
- [ ] replace all dangerous `open()` statements
- [ ] confirm the changes broke nothing

### CODE REVIEW

File: drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py [hunk 1]
```diff
@@ -38,16 +38,16 @@ def process(transaction):
...
-			try:
-				user = fileInfo["user"]
-			except:
-				user = None
-			secname = fileInfo["info"]
-			code = fileInfo["barcode"]
-			datasetType = fileInfo["type"]
+			with open(os.path.join(incomingPath, f)) as metadata
+        fileInfo = dict(line.strip().split('=') for line in metadata)
+        metadata.close()
```
Comment (wow-such-code) [inline drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py:43]:
```suggestion
        fileInfo = dict(line.strip().split('=') for line in metadata)
```
```suggestion
fileInfo = dict(line.strip().split('=') for line in metadata)
```
```suggestion
                                fileInfo = dict(line.strip().split('=') for line in metadata)
```

Comment (KochTobi) [reply drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py:43]: what is the difference in your suggestions apart from the indentation?

Comment (wow-such-code) [inline drop-boxes/register-attachments-dropbox/register-attachment-dropbox.py:43]:
```suggestion
                                fileInfo = dict(line.strip().split('=') for line in metadata)
```

File: drop-boxes/register-fasta-dropbox/register-fasta.py [hunk 2]
```diff
@@ -65,7 +62,7 @@ def process(transaction):
         project = identifier[:5]
         parentCode = identifier[:10]
     else:
-        print "The identifier "+identifier+" did not match the pattern Q[A-Z]{4}\d{3}\w{2} or checksum"
+        print("The identifier "+identifier+" did not match the pattern Q[A-Z]{4}\d{3}\w{2} or checksum")
```
Comment (wow-such-code) [inline drop-boxes/register-fasta-dropbox/register-fasta.py:65]: does this work in this version/make a difference?

Comment (KochTobi) [reply drop-boxes/register-fasta-dropbox/register-fasta.py:65]: My IDE probably added the parenthesis. I could not exactly test it so I do not know.

File: drop-boxes/register-fasta-dropbox/register-fasta.py [hunk 3]
```diff
@@ -138,11 +135,10 @@ def process(transaction):
...
             if f.endswith('origlabfilename'):
-                origName = open(os.path.join(incomingPath,f), 'r')
-                secondaryName = origName.readline().strip().split('_')[0]
-                origName.close()
-                sa.setPropertyValue('Q_SECONDARY_NAME', secondaryName)
-                os.remove(os.path.realpath(os.path.join(incomingPath,f)))
+                with open(os.path.join(incomingPath,f), 'r') as origName:
+                  secondaryName = origName.readline().strip().split('_')[0]
+                  sa.setPropertyValue('Q_SECONDARY_NAME', secondaryName)
+                  os.remove(os.path.realpath(os.path.join(incomingPath,f)))
```
Comment (wow-such-code) [inline drop-boxes/register-fasta-dropbox/register-fasta.py:141]: not sure you can remove it while it's open, i would put this (and maybe the line before) outside the "with" block

Comment (KochTobi) [reply drop-boxes/register-fasta-dropbox/register-fasta.py:141]: Yes

File: drop-boxes/register-imgag-dropbox/backup-imgag.py [hunk 4]
```diff
@@ -27,11 +27,9 @@
...
 
 def parse_metadata_file(filePath):
-	jsonFile = open(filePath, 'r')
-	data = json.load(jsonFile)
-	jsonFile.close()
-
-	return data
+	with open(filePath, 'r') as jsonFile:
+	  data = json.load(jsonFile)
+	  return data
```
Comment (wow-such-code) [inline drop-boxes/register-imgag-dropbox/backup-imgag.py:32]: would also put the return outside the "with" block

File: drop-boxes/register-imgag-dropbox/register-imgag.py [hunk 5]
```diff
@@ -33,11 +33,9 @@
...
 
 def parse_metadata_file(filePath):
-    jsonFile = open(filePath, 'r')
-    data = json.load(jsonFile)
-    jsonFile.close()
-
-    return data
+    with open(filePath, 'r') as jsonFile:
+      data = json.load(jsonFile)
+      return data
```
Comment (wow-such-code) [inline drop-boxes/register-imgag-dropbox/register-imgag.py:38]: return outside "with"

File: drop-boxes/register-fasta-dropbox/register-fasta.py (hunk 3 above)
Comment (KochTobi) [inline drop-boxes/register-fasta-dropbox/register-fasta.py:141]:
```suggestion
                with open(os.path.join(incomingPath,f), 'r') as origName:
                  secondaryName = origName.readline().strip().split('_')[0]
                sa.setPropertyValue('Q_SECONDARY_NAME', secondaryName)
                os.remove(os.path.realpath(os.path.join(incomingPath,f)))
```

File: drop-boxes/register-iontorrent-data/pushXML2cxxRest.py [hunk 6]
```diff
@@ -47,11 +47,11 @@ def pushXML2CxxREST(filepath):
...
     #files = {'file': io.open(filepath, 'r', encoding='utf8')}
-    xmlContent = codecs.open(filepath, encoding='utf-8')
-    #xmlContent = open(filepath, 'rb')
-    response = requests.post(importUrl, data=xmlContent, auth=restAuth, headers=headers, verify=False)
+    with codecs.open(filepath, encoding='utf-8') as xmlContent:
+      #xmlContent = open(filepath, 'rb')
+      response = requests.post(importUrl, data=xmlContent, auth=restAuth, headers=headers, verify=False)
 
-    return response
+      return response
```
Comment (wow-such-code) [inline drop-boxes/register-iontorrent-data/pushXML2cxxRest.py:54]: would return it outside the with

File: drop-boxes/register-iontorrent-data/register-iontorrent-data.py [hunk 7]
```diff
@@ -97,12 +97,12 @@ def create_barcode(projectCode, firstFreeBarcodeId, classChar):
...
-    while fileChunk:
-        sumObject.update(fileChunk)
-        fileChunk = infile.read(chunkSize)
-    return sumObject.hexdigest()
+    with open(fileFullPath, 'rb') as infile:
+      fileChunk = infile.read(chunkSize)
+      while fileChunk:
+          sumObject.update(fileChunk)
+          fileChunk = infile.read(chunkSize)
+      return sumObject.hexdigest()
```
Comment (wow-such-code) [inline drop-boxes/register-iontorrent-data/register-iontorrent-data.py:105]: return outside with

File: drop-boxes/register-iontorrent-data/register-iontorrent-data.py [hunk 8]
```diff
@@ -162,26 +162,23 @@ def findExperimentByID(expIdentifier, transaction):
...
-            break
-
-    return panelName
+    with open(fileName, 'r') as vcffile:
+      panelName = ''
+      for line in vcffile:
+          if line.startswith('##parametersName='):
+              panelName = line.strip().split('=')[1].strip('\"')
+              break
+      return panelName
```
Comment (wow-such-code) [inline drop-boxes/register-iontorrent-data/register-iontorrent-data.py:171]:
```suggestion
    panelName = ''
    with open(fileName, 'r') as vcffile:
      for line in vcffile:
          if line.startswith('##parametersName='):
              panelName = line.strip().split('=')[1].strip('\"')
              break
    return panelName
```

File: drop-boxes/register-qcml-dropbox/register-qcml-dropbox.py [hunk 9]
```diff
@@ -91,11 +91,10 @@ def process(transaction):
 
        	#cegat = False
         f = "source_dropbox.txt"
-       	sourceLabFile = open(os.path.join(incomingPath,f))
-        sourceLab = sourceLabFile.readline().strip()
-        sourceLabFile.close()
-        #if sourceLab == 'dmcegat':
-                #cegat = True
+       	sith open(os.path.join(incomingPath,f)) as sourceLabFile:
```
Comment (wow-such-code) [inline drop-boxes/register-qcml-dropbox/register-qcml-dropbox.py:94]: "Always two there are; no more, no less. A master and an apprentice."
```suggestion
       	with open(os.path.join(incomingPath,f)) as sourceLabFile:
```

File: drop-boxes/register-vcf-dropbox/register-vcf.py [hunk 10]
```diff
@@ -136,13 +136,13 @@ def process(transaction):
         vcSample.setExperiment(newVariantCallingExperiment) 
 
         cegat = False
-        sourceLabFile = open(os.path.join(incomingPath,'source_dropbox.txt'), 'r')
-        sourceLab = sourceLabFile.readline().strip()
-        sourceLabFile.close()
+        with open(os.path.join(incomingPath,'source_dropbox.txt'), 'r') as sourceLabFile:
+          sourceLab = sourceLabFile.readline().strip()
+          sourceLabFile.close()
```
Comment (wow-such-code) [inline drop-boxes/register-vcf-dropbox/register-vcf.py:141]:
```suggestion
```

File: drop-boxes/register-vcf-dropbox/register-vcf.py [hunk 11]
```diff
@@ -136,13 +136,13 @@ def process(transaction):
...
 
-        if sourceLab == 'dmcegat':
-            cegat = True
-        os.remove(os.path.realpath(os.path.join(incomingPath,'source_dropbox.txt')))
+          if sourceLab == 'dmcegat':
+              cegat = True
+         os.remove(os.path.realpath(os.path.join(incomingPath,'source_dropbox.txt')))
```
Comment (wow-such-code) [inline drop-boxes/register-vcf-dropbox/register-vcf.py:145]:
```suggestion
        os.remove(os.path.realpath(os.path.join(incomingPath,'source_dropbox.txt')))
```

**Review by wow-such-code — CHANGES_REQUESTED** [review]
looking good, made some requests, didn't fully review ion torrent, as it's not in use and quite complex

**Comment by KochTobi** [comment]
I will close this PR since it is to complex and outdated.

**Pull request closed without merging**